lib/circuitbox/memory_store.rb

```diff
@@ -0,0 +1,75 @@
+# frozen-string-literal: true
+
+require_relative 'memory_store/compactor'
+require_relative 'memory_store/container'
+
+class Circuitbox
+  class MemoryStore
+    include MonotonicTime
+
+    def initialize(compaction_frequency: 10)
+      @store = {}
```

Comment:
concurrent ruby's thread safe hash is just ```::Hash``` on MRI because of how the GVL works. It looks like on other ruby implementations it's wrapping the methods with a lock.

Even if this were switched to a thread safe hash there is still a need for a lock for methods like ```increment``` because of the need to load an existing value, update expiry, write new value.